src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs

```diff
@@ -124,14 +124,12 @@ public virtual SdkResult ResolveSdk(int submissionId, SdkReference sdk, LoggingC
                     // than give them a generic error, we want to give a more specific message.  This exception cannot be caught by
                     // the resolver itself because it is usually thrown before the class is loaded
                     // MSB4243: The NuGet-based SDK resolver failed to run because NuGet assemblies could not be located.  Check your installation of MSBuild or set the environment variable "{0}" to the folder that contains the required NuGet assemblies. {1}
-                    loggingContext.LogWarning(null, new BuildEventFileInfo(sdkReferenceLocation), "CouldNotRunNuGetSdkResolver", MSBuildConstants.NuGetAssemblyPathEnvironmentVariableName, e.Message);
-                    continue;
+                    throw new Exception(string.Format(ResourceUtilities.GetResourceString("CouldNotRunNuGetSdkResolver"), MSBuildConstants.NuGetAssemblyPathEnvironmentVariableName, e.Message));
                 }
                 catch (Exception e)
                 {
-                    // MSB4242: The SDK resolver "{0}" failed to run. {1}
-                    loggingContext.LogWarning(null, new BuildEventFileInfo(sdkReferenceLocation), "CouldNotRunSdkResolver", sdkResolver.Name, e.Message);
-                    continue;
+                    // MSB4242: The SDK resolver "{0}" failed while attempting to resolve the SDK "{1}": {2}
+                    throw new Exception(string.Format(ResourceUtilities.GetResourceString("SDKResolverFailed"), sdkResolver.Name, sdk.ToString(), e.Message));
```

Comment:
I know this wasn't in the warning, but maybe include all of e as an inner exception? (Also elsewhere) Then we can stack trace, etc.

Also, should the Exception (including above) be more specific?

Reply:
> I know this wasn't in the warning, but maybe include all of e as an inner exception? (Also elsewhere) Then we can stack trace, etc.

Makes sense